renderlib/Manipulator.cpp

```diff
@@ -0,0 +1,574 @@
+#include "Manipulator.h"
+
+// #include "AppScene.h"
+
+float ManipulationTool::s_manipulatorSize = 500.0f;
+
+void
+drawCircle(Gesture& gesture,
+           glm::vec3 center,
+           glm::vec3 xaxis,
+           glm::vec3 yaxis,
+           uint32_t numSegments,
+           glm::vec3 color,
+           float opacity,
+           uint32_t code)
+{
+  for (int i = 0; i < numSegments; ++i) {
+    float t0 = float(i) / float(numSegments);
+    float t1 = float(i + 1) / float(numSegments);
+
+    float theta0 = t0 * 2.0f * glm::pi<float>();
+    float theta1 = t1 * 2.0f * glm::pi<float>();
+
+    glm::vec3 p0 = center + xaxis * cosf(theta0) + yaxis * sinf(theta0);
+    glm::vec3 p1 = center + xaxis * cosf(theta1) + yaxis * sinf(theta1);
+
+    gesture.graphics.addLine(Gesture::Graphics::VertsCode(p0, color, opacity, code),
+                             Gesture::Graphics::VertsCode(p1, color, opacity, code));
+  }
+}
+
+// does not draw a flat base
+void
+drawCone(Gesture& gesture,
```

Comment:
Maybe worth commenting that the `xaxis`, `yaxis`, `zaxis` variables also control height/radius of the cone?